### PR TITLE
Allow tracking users with no keys

### PR DESCRIPTION
**Pull request by patrickxb**
This patch allows tracking users with no keys and adds a test to ensure that after the user gets device keys via provisioning, the identify ui displays a new eldest key notice.

### CODE REVIEW

File: go/engine/track.go [hunk 1]
```diff
@@ -71,7 +71,9 @@ func (e *TrackEngine) Run(ctx *Context) error {
 
 	upk := ieng.Result().Upk
 	var err error
-	e.them, err = libkb.LoadUser(libkb.NewLoadUserByUIDArg(ctx.GetNetContext(), e.G(), upk.Uid))
+	loadarg := libkb.NewLoadUserByUIDArg(ctx.GetNetContext(), e.G(), upk.Uid)
```
Comment (maxtaco) [inline go/engine/track.go]: Make a helper for this --- `NewLoadUserByUIDOptionalPublicKeyArg` ?

File: go/libkb/kbsig.go [hunk 2]
```diff
@@ -150,7 +150,13 @@ func (u *User) ToTrackingStatementSeqTail() *jsonw.Wrapper {
 func (u *User) ToTrackingStatement(w *jsonw.Wrapper, outcome *IdentifyOutcome) (err error) {
 
 	track := jsonw.NewDictionary()
-	track.SetKey("key", u.ToTrackingStatementKey(&err))
+	var keyErr error
+	key := u.ToTrackingStatementKey(&keyErr)
```
Comment (maxtaco) [inline go/libkb/kbsig.go]: Can this be cleaned up a bit?  I think it's better to not call this if `!u.HasActiveKey()`?

**Comment by patrickxb** [comment]
Did you see the test?  It does what you specify in your bullet points.

**Comment by maxtaco** [comment]
I did see the test, but that test wouldn't cover what the user sees in the UI.

File: go/libkb/track.go [hunk 3]
```diff
@@ -363,6 +363,9 @@ func (t TrackDiffNewEldest) GetTrackDiffType() keybase1.TrackDiffType {
 	return keybase1.TrackDiffType_NEW_ELDEST
 }
 func (t TrackDiffNewEldest) ToDisplayString() string {
+	if t.tracked.IsNil() {
+		return fmt.Sprintf("No key when tracked; established new eldest key %s", t.observed)
```
Comment (maxtaco) [inline go/libkb/track.go]: s/tracked/followed/

**Pull request merged**